internal/experiment/generation/transformer.go

```diff
@@ -236,47 +237,22 @@ func (p *pnode) TargetRef() *corev1.ObjectReference {
 
 // parameterNamer returns a name generation function for parameters based on scan results.
 func parameterNamer(selected []interface{}) ParameterNamer {
-	// Index the object references by kind and name
-	type targeted interface {
-		TargetRef() *corev1.ObjectReference
-	}
-	needsPath := make(map[string]map[string]int)
-	for _, sel := range selected {
-		t, ok := sel.(targeted)
-		if !ok {
-			continue
-		}
-		targetRef := t.TargetRef()
-		if ns := needsPath[targetRef.Kind]; ns == nil {
-			needsPath[targetRef.Kind] = make(map[string]int)
-		}
-		needsPath[targetRef.Kind][targetRef.Name]++
-	}
-
-	// Determine which prefixes we need
-	needsKind := len(needsPath) > 1
-	needsName := false
-	for _, v := range needsPath {
-		needsName = needsName || len(v) > 1
-	}
-
 	return func(meta yaml.ResourceMeta, path []string, name string) string {
-		var parts []string
-
-		if needsKind {
-			parts = append(parts, meta.Kind)
-		}
-
-		if needsName {
-			parts = append(parts, meta.Name)
+		parts := []string{
+			meta.Kind,
+			meta.Name,
 		}
 
-		if needsPath[meta.Kind][meta.Name] > 1 {
-			for _, p := range path {
-				if yaml.IsListIndex(p) {
-					if _, value, _ := yaml.SplitIndexNameValue(p); value != "" {
-						parts = append(parts, value)
+		// In this instance we have multiple containers in a pod that we're
+		// dealing with, so we need to add container name to the parameter
+		// name to better namespace it
+		for pathIdx, p := range path {
+			if yaml.IsListIndex(p) {
+				if _, value, _ := yaml.SplitIndexNameValue(p); value != "" {
+					if path[pathIdx-1] == "env" {
+						parts = append(parts, "env")
```

Comment:
I don't think we need this, we didn't have it before. As long as the `name` comes in blank we'll just end up with `memory`, `cpu`, `replicas`, `MY_ENV_VAR` all at the same level in the hierarchy (i.e. there is no `env` magic value equivalent in the name hierarchy for container resources or replicas).

Reply:
When I was thinking through this one, I got a grin like this smug face -- :trollface: 

There's no hard requirement that an environment variable be capitalized, so it's possible someone may use `cpu` or `memory` or `replicas` as the environment variable in addition to wanting to tune requests/resources and we'd wind up with an invalid experiment because of duplicated parameters.

Also I was thinking by injecting `env` there, it'd also make it better for display purposes in the UI .

Reply:
What if they have a container named `env`? :trollface:

Reply:
💔

Reply:
Interestingly enough, I couldn't break it with container name == env, so yay for that.